metagraph/src/cli/query.cpp

```diff
@@ -39,122 +41,336 @@ using mtg::graph::boss::BOSSConstructor;
 typedef typename mtg::graph::DeBruijnGraph::node_index node_index;
 
 
-QueryExecutor::QueryExecutor(const Config &config,
-                             const graph::AnnotatedDBG &anno_graph,
-                             std::unique_ptr<graph::align::DBGAlignerConfig>&& aligner_config,
-                             ThreadPool &thread_pool)
-      : config_(config),
-        anno_graph_(anno_graph),
-        aligner_config_(std::move(aligner_config)),
-        thread_pool_(thread_pool) { }
-
-std::string QueryExecutor::execute_query(const std::string &seq_name,
-                                         const std::string &sequence,
-                                         bool count_labels,
-                                         bool print_signature,
-                                         bool suppress_unlabeled,
-                                         size_t num_top_labels,
-                                         double discovery_fraction,
-                                         double presence_fraction,
-                                         std::string anno_labels_delimiter,
-                                         const AnnotatedDBG &anno_graph,
-                                         bool with_kmer_counts,
-                                         const std::vector<double> &count_quantiles,
-                                         bool query_coords) {
-    std::string output;
-    output.reserve(1'000);
+/**
+ * Given a vector of kmer label matched coordinates, collapse continuous ranges of coordinates
+ * to start-end tuples.
+ *
+ * @param coords    the vector of tuples as stored in a SeqSearchResult
+ *                  originally returned from AnnotatedDBG::get_kmer_coordinates
+ * @return vector of 'begin-end' range string representations
+ */
+std::vector<std::string>
+get_collapsed_coord_ranges(const std::vector<SmallVector<uint64_t>> &tuples) {
+    // Build output
+    std::vector<std::string> ranges;
+
+    // Keep track of the beginning of the range (just use a sentinel)
+    uint64_t begin = 0;
+    uint64_t current = 0;
+    bool in_range = false;
+
+    for (const auto &coords : tuples) {
+        if (coords.empty()) continue;
+
+        // TODO: Look for example where row_tuples are multiple values?
```

Comment:
What do you mean?

Reply:
oh, never mind. I see what you mean...

Reply:
You can think of the case where the indexed sequence is AAA...A and the query is AAA.
So essentially every k-mer gets all the coordinates.

The current implementation will only print one coordinate range, which is incorrect. So this should be fixed. But we should agree on the desired output format first.